### PR TITLE
chore(ci): update circleci config to use current images

### PR DESCRIPTION
**Pull request by louh**
[Link to issue.](https://github.com/usdoj-crt/crt-portal-management/issues/1153)

## What does this change?

This updates our CircleCI configuration to migrate to "next-generation" build images, as described here: https://circleci.com/docs/2.0/next-gen-migration-guide/

- Legacy `circleci/python:3.9.7-node-browsers` images have been updated to `cimg/python:3.9.7-browsers`
- Configuration has been updated to specify version 2.1 in case we use orbs. However, we currently don't need it.
- Because the new images have a different Python environment than the legacy images, previously cached Python dependencies broke the CI job. Starting with this update, we follow the pattern in CircleCI's [Caching Dependencies documentation](https://circleci.com/docs/2.0/caching/#clearing-cache) to hard-code version numbers in cache keys rather than using project-wide environment variables.

## Screenshots (for front-end PR):

n/a

### Author

+ [x] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [x] Check for, document, and establish a testing plan for any behavior that may vary across environments or is otherwise difficult to test.
+ [x] Check for [accessibility](https://github.com/usdoj-crt/crt-portal/blob/develop/docs/a11y_plan.md).
+ [x] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

### Reviewer

+ [x] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [x] Check for any behavior that may vary across environments or is difficult to test, and ensure that it is well-understood, documented, and that there is a testing plan in place.
+ [x] Re-check for [accessibility](https://github.com/usdoj-crt/crt-portal/blob/develop/docs/a11y_plan.md).
+ [x] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

## Notes for reviewer:

See [PR instructions doc](https://github.com/usdoj/crt-portal/blob/master/docs/pull_requests.md) for full pull request review instructions.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -205,14 +206,12 @@ jobs:
...
           POSTGRES_DB: circle_test
-
-
     steps:
       - checkout
       # removing as we deal with caching problem
       # - restore_cache:
       # # Read about caching dependencies: https://circleci.com/docs/2.0/caching/
-      #     key: deps-{{ .Environment.CACHE_VERSION }}-{{ .Branch }}-{{ checksum "Pipfile.lock" }}-{{ checksum "package.json" }}
+      #     key: deps-v4-{{ .Branch }}-{{ checksum "Pipfile.lock" }}-{{ checksum "package.json" }}
```
Comment (billyhunt) [inline .circleci/config.yml:214]: Do we need this commented out code?

**Review by billyhunt — APPROVED** [review]
Looks good to me!

File: .circleci/config.yml (hunk 1 above)
Comment (louh) [reply .circleci/config.yml:214]: I haven't looked at what this does, but I'm okay with leaving it alone if you are.

Comment (billyhunt) [reply .circleci/config.yml:214]: Works for me!

**Pull request merged**